doc/internal-api.rst

```diff
@@ -66,19 +67,52 @@ methods:
 
 
 
+About the IO module
+-------------------
+
+Handling of the different file formats is implemented in :mod:`can.io`.
+Each file/IO type is within a separate module and ideally implements both a *Reader* and a *Writer*.
+The reader usually extends :class:`can.io.generic.BaseIOHandler`, while
+the writer additionally extends :class:`can.Listener`,
+to be able to be passed directly to a :class:`can.Notifier`.
+
+
+
+Adding support for new file formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This assumes that you want to add a new file format, called *canstore*.
+
+1. Create a new module: *can/io/canstore.py*
+   (*or* simply copy some existing one like *can/io/csv.py*)
+2. Implement a reader ``CanstoreReader`` (which extends :class:`can.io.generic.BaseIOHandler`).
```

Comment:
A `Reader` doesn't *have* to extend `BaseIOHandler`, that is a good starting point for any *file* based format though.

A small suggestion is to add a `~` to the path to improve readability in the rendered docs (but ensuring the link works).

```suggestion
2. Implement a reader ``CanstoreReader`` (which extends :class:`~can.io.generic.BaseIOHandler`).
```

Reply:
I thought that it might be nice to know where this class sits, since this is documentation for the library developers.